ledger/participant-integration-api/src/test/lib/scala/platform/store/PersistentStoreSpecBase.scala

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store
+
+import java.util.concurrent.Executors
+import java.util.concurrent.atomic.AtomicInteger
+
+import com.codahale.metrics.MetricRegistry
+import com.daml.ledger.resources.ResourceContext
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.metrics.{DatabaseMetrics, Metrics}
+import com.daml.platform.configuration.ServerRole
+import com.daml.platform.store.DbSupport.{ConnectionPoolConfig, DbConfig}
+import com.daml.platform.store.backend.StorageBackendProvider
+import com.daml.resources.Resource
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach, Suite}
+
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.concurrent.duration._
+
+/** Base class for running persistent store-level tests.
+  *
+  * Features:
+  * - Before all test cases creates a new database and applies the flyway migrations to it.
+  * - Before each test case resets the contents of the database.
+  * - Ensures that at most one test case runs at a time.
+  */
+trait PersistentStoreSpecBase
+    extends BeforeAndAfterEach
+    with BeforeAndAfterAll
+    with StorageBackendProvider { this: Suite =>
+
+  implicit protected val loggingContext: LoggingContext = LoggingContext.ForTesting
+
+  protected val logger: ContextualizedLogger = ContextualizedLogger.get(getClass)
+
+  private val runningTests = new AtomicInteger(0)
+  private val thisSimpleName = getClass.getSimpleName
+
+  protected var dbSupport: DbSupport = _
+
+  // Each test should start with an empty database to allow testing low-level behavior
+  // However, creating a fresh database for each test would be too expensive.
+  // Instead, we truncate all tables using the reset() call before each test.
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    val dbMetrics = DatabaseMetrics.ForTesting(getClass.getSimpleName)
+    val resetDbF = dbSupport.dbDispatcher.executeSql(dbMetrics) { connection =>
+      backend.reset.resetAll(connection)
+      updateLedgerEndCache(connection)
```

Comment:
It seems to me fishy, that you need to care about ledger end and string interning in here. These are indexer concepts. I assuming the stores you would like to test should be independent from indexer, no?

Reply:
I copied the approach from `com.daml.platform.store.backend.StorageBackendSpec`. Not sure if it's necessary here